pkg/pillar/cmd/baseosmgr/handlebaseos.go

```diff
@@ -1046,10 +1046,9 @@ func updateBaseOsStatusOnReboot(ctxPtr *baseOsMgrContext) {
 
 // first pick up from the partition
```

Comment:
This is quite subtle (and the comment is out of date)
I think the needed comment is
 - assumes the callers verify that the other partition is "inprogress" which means we most recently booted that partition
But that might be false, because after a fallback the other partition is marked "inprogress", and then if we reboot again the other partition is still marked "inprogress" yet the most recent reboot reason is due to the reboot from active to active.

So you need to track which image/partition was running when the reboot happened.